lib/guest_installation_and_configuration_base.pm

```diff
@@ -0,0 +1,1565 @@
+    $_bridge_forwarddelay //= '15';
+    script_run("cp /etc/sysconfig/network/ifcfg-$_name /etc/sysconfig/network/backup-ifcfg-$_name");
+    script_run("cp /etc/sysconfig/network/backup-ifcfg-$_name $common_log_folder");
+    my $_bridge_device_config_file = '/etc/sysconfig/network/ifcfg-' . $_name;
+    $_ipaddr = '0.0.0.0' if ($_ipaddr eq '');
+    $_ipaddr = ''        if (!($_ipaddr =~ /\d+\.\d+\.\d+\.\d+/));
+    type_string("cat > $_bridge_device_config_file <<EOF
+IPADDR=\'$_ipaddr\'
+NAME=\'$_name\'
+BOOTPROTO=\'$_bootproto\'
+STARTMODE=\'$_startmode\'
+ZONE=\'$_zone\'
+BRIDGE=\'$_bridge\'
+BRIDGE_PORTS=\'$_bridge_ports\'
+BRIDGE_STP=\'$_bridge_stp\'
+BRIDGE_FORWARDDELAY=\'$_bridge_forwarddelay\'
+EOF
+");
+    script_run("cp $_bridge_device_config_file $common_log_folder");
+    record_info("Network device $_name config $_bridge_device_config_file", script_output("cat $_bridge_device_config_file", proceed_on_failure => 0));
+    return $self;
+}
+
+#Create [guest_network_device] by writing device information into ifcfg file in /etc/sysconfig/network.Mark guest installation as FAILED if [guest_network_device] can not be
+#successfully started up.If [guest_network_device] or [guest_netaddr] already exists and active on host judging by "ip route show",both of them will not be created anyway.
+sub config_guest_network_bridge_device {
+    my $self = shift;
+
+    $self->reveal_myself;
+    my $_bridge_network          = shift;
+    my $_bridge_network_in_route = shift;
+    my $_bridge_device           = shift;
+    $_bridge_device //= $self->{guest_network_device};
+    if ((script_output("ip route show | grep -o $_bridge_device", proceed_on_failure => 1) eq '') and (script_output("ip route show | grep -o $_bridge_network_in_route", proceed_on_failure => 1) eq '')) {
+        my $_detect_active_route   = '';
+        my $_detect_inactive_route = '';
+        if ($self->{guest_netaddr} ne 'host-default') {
+            $self->write_guest_network_bridge_device_config($_bridge_network, $_bridge_device, 'static', 'auto', '', 'yes');
+            my $_bridge_device_config_file = '/etc/sysconfig/network/ifcfg-' . $_bridge_device;
+            script_retry("wicked ifup $_bridge_device_config_file $_bridge_device", retry => 3, die => 0);
+            $_detect_active_route = script_output("ip route show | grep -i $_bridge_device", proceed_on_failure => 1);
+        }
+        else {
+            my $_host_default_network_interface = script_output("ip route show default | grep -i dhcp | grep -vE br[[:digit:]]+ | head -1 | awk \'{print \$5}\'");
+            $self->write_guest_network_bridge_device_config($_bridge_network, $_bridge_device, 'dhcp', 'auto', '', 'yes', $_host_default_network_interface);
+            $self->write_guest_network_bridge_device_config('', $_host_default_network_interface, 'none');
+            script_retry("systemctl restart network", timeout => 60, delay => 15, retry => 3, die => 0);
+            type_string("reset\n");
```

Comment:
enter_cmd(), a wrapper of type_string(), is newly introduced. You can consider it in your script.

Reply:
I will start picking up this enter_cmd() in the future if this is useful. Keep use familiar type_string() here.